src/test/failure_reporter.h

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef TEST_FAILURE_REPORTER_H
+#define TEST_FAILURE_REPORTER_H
+
+#include <beast/unit_test/amount.hpp>
+#include <beast/unit_test/recorder.hpp>
+#include <boost/lexical_cast.hpp>
```

Comment:
I couldn't see any obvious reason to include the following:

```
#include <boost/lexical_cast.hpp>
#include <boost/optional.hpp>
#include <algorithm>
#include <chrono>
#include <functional>
#include <sstream>
#include <iomanip>
```
...so I did a nonunity build with these removed and it worked fine. You might consider removing them.